packages/contracts/test/exchange/core_internal.ts

```diff
@@ -0,0 +1,134 @@
+    let feeRecipientAddress: string;
+
+    let erc20TokenA: DummyERC20TokenContract;
+    let erc20TokenB: DummyERC20TokenContract;
+    let zrxToken: DummyERC20TokenContract;
+    let erc20Proxy: ERC20ProxyContract;
+
+    let signedOrder: SignedOrder;
+    let erc20Wrapper: ERC20Wrapper;
+    let orderFactory: OrderFactory;
+
+    let defaultMakerAssetAddress: string;
+    let defaultTakerAssetAddress: string;
+    before(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    before(async () => {
+        testExchange = await TestMixinExchangeCoreContract.deployFrom0xArtifactAsync(
+            artifacts.TestMixinExchangeCore,
+            provider,
+            txDefaults,
+        );
+
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        const usedAddresses = ([owner, makerAddress, takerAddress, feeRecipientAddress] = _.slice(accounts, 0, 4));
+
+        erc20Wrapper = new ERC20Wrapper(provider, usedAddresses, owner);
+
+        const numDummyErc20ToDeploy = 3;
+        [erc20TokenA, erc20TokenB, zrxToken] = await erc20Wrapper.deployDummyTokensAsync(
+            numDummyErc20ToDeploy,
+            constants.DUMMY_TOKEN_DECIMALS,
+        );
+        erc20Proxy = await erc20Wrapper.deployProxyAsync();
+        await erc20Wrapper.setBalancesAndAllowancesAsync();
+
+        await web3Wrapper.awaitTransactionSuccessAsync(
+            await erc20Proxy.addAuthorizedAddress.sendTransactionAsync(testExchange.address, {
+                from: owner,
+            }),
+            constants.AWAIT_TRANSACTION_MINED_MS,
+        );
+
+        defaultMakerAssetAddress = erc20TokenA.address;
+        defaultTakerAssetAddress = erc20TokenB.address;
+
+        const defaultOrderParams = {
+            ...constants.STATIC_ORDER_PARAMS,
+            exchangeAddress: testExchange.address,
+            makerAddress,
+            feeRecipientAddress,
+            makerAssetData: assetProxyUtils.encodeERC20AssetData(defaultMakerAssetAddress),
+            takerAssetData: assetProxyUtils.encodeERC20AssetData(defaultTakerAssetAddress),
+        };
+        const privateKey = constants.TESTRPC_PRIVATE_KEYS[accounts.indexOf(makerAddress)];
+        orderFactory = new OrderFactory(privateKey, defaultOrderParams);
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+
+    describe.only('updateFilledState', async () => {
+        const testCases = combinatorics.cartesianProduct(
+            positiveNaturalBigNumbers,
+            positiveNaturalBigNumbers,
+            positiveNaturalBigNumbers,
+            positiveNaturalBigNumbers,
+        );
+        console.log(`Generated ${testCases.length} combinatoric test cases.`);
+        let counter = -1;
+        testCases.forEach(async testCase => {
```

Comment:
Each test case consists of four numeric values made up of different combinations of the numbers in `positiveNaturalBigNumbers`. For each case, we simply pass the values through to `sendTransactionAsync` and make sure the transaction is successful.